blocks/block-edit/context.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import { Component, createContext, createHigherOrderComponent } from '@wordpress/element';
+
+const { Consumer, Provider } = createContext( {
+	isSelected: true,
+} );
+
+/**
+ * A Higher Order Component used to be provide a context for BlockEdit
+ * component.
+ *
+ * @param {Component} OriginalComponent Component to wrap.
+ *
+ * @return {Component} Component with a BlockEdit context set.
+ */
+export const withBlockEditContextProvider = createHigherOrderComponent( ( OriginalComponent ) => {
```

Comment:
I don't like the fact that Provider is included using a HoC instead of a regular Component using children. Because it gives the impression that we could use this component in several places where in reality the providers are used only once.

Thoughts?

Reply:
Good point made :)
I will move it to `BlockEdit`.

> Because it gives the impression that we could use this component in several places where in reality the providers are used only once.

Well, not in every case, but in general I agree 👍